components/result/run-history_test.go

```diff
@@ -0,0 +1,413 @@
+			//But the chaos should be aborted before hand
+			By("[Status]: Wait for chaos pod completion and then print logs")
+			err = pkg.ChaosPodLogs(&testsDetails, clients)
+			Expect(err).To(BeNil(), "Failed to get the experiment chaos pod logs, due to {%v}", err)
+
+			//Checking chaosengine verdict
+			By("Checking the Verdict of Chaos Engine")
+			chaosEngineVerdict, err := pkg.GetChaosEngineVerdict(&testsDetails, clients)
+			Expect(err).To(BeNil(), "Fail to get the chaosengine Verdict, due to {%v}", err)
+			Expect(chaosEngineVerdict).To(Equal("Fail"), "ChaosEngine Verdict is not Stopped, due to {%v}", err)
+
+			// Checking the chaosresult run history verdict
+			By("[Verdict]: CHecking the chaosresult run history verdict")
+			err = pkg.ChaosRunHistoryVerdict(&testsDetails, clients, runHistory)
+			Expect(err).To(BeNil(), "ChaosResult Run History Check failed, due to {%v}", err)
+
+		})
+	})
+
+	Context("Run history check for stopped run", func() {
+
+		It("Should check for the run history in the chaos result when experiment is aborted", func() {
+
+			testsDetails := types.TestDetails{}
+			clients := environment.ClientSets{}
+			chaosExperiment := v1alpha1.ChaosExperiment{}
+			chaosEngine := v1alpha1.ChaosEngine{}
+
+			//Getting kubeConfig and Generate ClientSets
+			By("[PreChaos]: Getting kubeconfig and generate clientset")
+			err := clients.GenerateClientSetFromKubeConfig()
+			Expect(err).To(BeNil(), "Unable to Get the kubeconfig, due to {%v}", err)
+
+			//Fetching all the default ENV
+			By("[PreChaos]: Fetching all default ENVs")
+			klog.Infof("[PreReq]: Getting the ENVs for the %v test", testsDetails.ExperimentName)
+			environment.GetENV(&testsDetails, "pod-delete", "run-history-stopped1")
+			// environment.GetENV(&testsDetails, "pod-delete", "run-history")
+
+			// Checking the chaos operator running status
+			By("[Status]: Checking chaos operator status")
+			err = pkg.OperatorStatusCheck(&testsDetails, clients)
+			Expect(err).To(BeNil(), "Operator status check failed, due to {%v}", err)
+
+			// Get Run History
+			runHistory := pkg.GetRunHistoryStatus(&testsDetails, clients)
+
+			// Prepare Chaos Execution
+			By("[Prepare]: Prepare Chaos Execution")
+			err = pkg.PrepareChaos(&testsDetails, &chaosExperiment, &chaosEngine, clients, false)
+			Expect(err).To(BeNil(), "fail to prepare chaos, due to {%v}", err)
+
+			//Checking runner pod running state
+			By("[Status]: Runner pod running status check")
+			_, err = pkg.RunnerPodStatus(&testsDetails, testsDetails.AppNS, clients)
+			Expect(err).To(BeNil(), "Runner pod status check failed, due to {%v}", err)
+
+			//Chaos pod running status check
+			err = pkg.ChaosPodStatus(&testsDetails, clients)
+			Expect(err).To(BeNil(), "Chaos pod status check failed, due to {%v}", err)
+
+			//Waiting for chaosresult creation from experiment
+			klog.Info("[Wait]: waiting for chaosresult creation from experiment")
+			time.Sleep(15 * time.Second)
```

Comment:
```15``` Is it hardcoded needed?

Reply:
This is for waiting some time before aborting the experiment. 
cc: @uditgaurav

Reply:
This should not be hardcoded. We can make it dynamic by continuously checking the experiment status and abort in between.
But it can also be done in a different PR as it is present in multiple places.